packages/mui-joy/src/Select/Select.tsx

```diff
@@ -240,6 +241,8 @@ const SelectListbox = styled(StyledList, {
     '--List-item-stickyTop': 'calc(var(--List-padding, var(--List-divider-gap)) * -1)', // negative amount of the List's padding block
     ...scopedVariables,
     minWidth: 'max-content', // prevent options from shrinking if some of them is wider than the Select's root.
+    maxHeight: '44vh', // the best value from what I tried so far which does not cause screen flicker when listbox is open.
```

Comment:
IMHO we should take a lot more vertical space, Radix feels better on my 14" screen:

<img width="609" alt="Screenshot 2023-08-23 at 11 51 48" src="https://github.com/mui/material-ui/assets/3165635/98045a20-f733-428e-820c-8eabcc806c53">

Same as macOS and Material UI:

<img width="181" alt="Screenshot 2023-08-23 at 11 53 12" src="https://github.com/mui/material-ui/assets/3165635/32b6883e-3025-444c-b694-1c4643debe4b">

<img width="266" alt="Screenshot 2023-08-23 at 11 52 36" src="https://github.com/mui/material-ui/assets/3165635/3e27c5a4-3f8f-481b-8e19-9a56f27b8d13">

Reply:
Issue created https://github.com/mui/material-ui/issues/38610